streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceivedBlockTracker.scala

```diff
@@ -73,6 +79,18 @@ private[streaming] class ReceivedBlockTracker(
   private val streamIdToUnallocatedBlockQueues = new mutable.HashMap[Int, ReceivedBlockQueue]
   private val timeToAllocatedBlocks = new mutable.HashMap[Time, AllocatedBlocks]
   private val writeAheadLogOption = createWriteAheadLog()
+  // exposed for tests
+  protected val walWriteQueue = new LinkedBlockingQueue[ReceivedBlockTrackerLogEvent]()
```

Comment:
Rather than have the corresponding promise in another map, why not just keep the event+promise in the queue? Wouldnt it be a simpler design?